### DISCUSSION TITLE
Trouble with large file sizes

### DISCUSSION THREAD

**Discussion by dizhydron** [HaxeFlixel/flixel#2637] [Q&A] [answered]
Hi! Novice to programming here messing around with HaxeFlixel. Trying to find information about the upper limit of what this engine can process when loading a single image. I've had luck with images up to 26mb in size but get crashes for things much larger than that testing around the 100mb mark. I'm assuming that it's a RAM issue. Whatever default ram allocation for a HaxeFlixel project is too low for that.

Is there a max file size that an image can be to be loaded into HaxeFlixel? Is there any way to allocate more RAM to process larger images? I know that you can kind of do that in Task manager but as far as I know that shouldn't actually work for what the engine can process.

**Comment by Geokureli** [comment, marked as answer]
flixel's limits are really openfl limits, becuase flixel uses openfl as a backend. And yes I do believe there are are hard limits on image sizes, a quick search results in:
https://api.openfl.org/openfl/display/BitmapData.html
> a bitmap can consume up to 64 MB of memory
and it probably depends on the target

**Reply by dizhydron** [reply]
Thanks for linking this. I'll take a look. Anyway I might possibly be able to get around the 64 mb limit that you can think of?

**Reply by Geokureli** [reply]
> Anyway I might possibly be able to get around the 64 mb limit that you can think of?

Nah, you would have to ask in the OpenFL forum
